components/ai_chat/resources/page/state/conversation_context.tsx

```diff
@@ -7,16 +7,19 @@ import * as React from 'react'
 
 import * as mojom from 'gen/brave/components/ai_chat/core/common/mojom/ai_chat.mojom.m.js'
 import usePromise from '$web-common/usePromise'
-import * as API from '../api/'
+import getAPI, * as API from '../api/'
 import { useAIChat } from './ai_chat_context'
 import { isLeoModel } from '../model_utils'
+import { useSelectedConversation } from '../routes'
+import useIsConversationVisible from '../hooks/useIsConversationVisible'
 
 const MAX_INPUT_CHAR = 2000
 const CHAR_LIMIT_THRESHOLD = MAX_INPUT_CHAR * 0.8
 
 export interface ConversationContextProps {
   conversationHandler: API.ConversationHandlerRemote
   callbackRouter: API.ConversationUICallbackRouter
+  selectedConversationId: string | undefined
```

Comment:
I think ideally we'd keep this out of the ConversationContext and both it and the knowledge of URLs / selected conversation out of this file. I can imagine we're not far off having conversations in other UIs (e.g. popups) outside of the AI Chat full-page / sidebar. We're probably more farther off having more than 1 conversation displayed at a time, but it's possible... All that to say it seems like URL / selected conversation should be handled at a higher, "page", level rather than a component responsible for only part of the page state/UI. Whether it's AIChatContext or something new like AIChatPageContext 🤷 
What do you think?

Reply:
Sweet, added a new ActiveConversationProvider :smile: